src/classes/BDI_DataImportService.cls

```diff
@@ -38,9 +38,19 @@
 *
 */
 global with sharing class BDI_DataImportService {
+    private static DescribeFieldResult batchNumberDescribe = UTIL_Describe.getFieldDescribe(
+        String.valueOf(DataImport__c.SObjectType),
+        String.valueOf(DataImport__c.NPSP_Data_Import_Batch__c)
+    );
+
+    public static final String BATCH_NUMBER_QUERY_STR = String.join(new List<String>{
```

Comment:
This is relatively minor, but for the naming of this variable - this string isn't a complete query string, just the dot-notated field from data import - so maybe this could be named "RELATED_BATCH_NUMBER_FIELD" or "DI_BATCH_NUMBER_FIELD" or even "DATAIMPORT_BATCH_NUMBER_FIELD"?

Reply:
fixed!